src/map/battle.c

```diff
@@ -4855,15 +4860,15 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 
 	if (sd && !skill_id) {
 		//Check for double attack.
-		if (( (skill_lv=pc->checkskill(sd,TF_DOUBLE)) > 0 && sd->weapontype1 == W_DAGGER )
+		if (( (skill_lv = pc->checkskill(sd,TF_DOUBLE)) > 0 && sd->weapontype1 == W_DAGGER )
 		 || ( sd->bonus.double_rate > 0 && sd->weapontype1 != W_FIST ) //Will fail bare-handed
 		 || ( sc && sc->data[SC_KAGEMUSYA] && sd->weapontype1 != W_FIST ) // Need confirmation
 		) {
 			//Success chance is not added, the higher one is used [Skotlex]
-			if( rnd()%100 < ( 5*skill_lv > sd->bonus.double_rate ? 5*skill_lv : sc && sc->data[SC_KAGEMUSYA]?sc->data[SC_KAGEMUSYA]->val1*3:sd->bonus.double_rate ) )
-			{
-				wd.div_ = skill->get_num(TF_DOUBLE,skill_lv?skill_lv:1);
+			if (rnd() % 100 < (5 * skill_lv > sd->bonus.double_rate ? 5 * skill_lv : sc && sc->data[SC_KAGEMUSYA] ? sc->data[SC_KAGEMUSYA]->val1 * 3 : sd->bonus.double_rate)) {
+				wd.div_ = skill->get_num(TF_DOUBLE, skill_lv ? skill_lv : 1);
 				wd.type = BDT_MULTIHIT;
+				hitrate += skill_lv;
```

Comment:
That's not how it works in pre-renewal.

The bonus isn't a flat +10 HIT.

It's added up together with the BS_WEAPONRESEARCH bonus and potentially other percentual ones and then the hitrate stemming from own hit, enemy flee, fogwall and other status effects gets this as a percentual bonus.

So it's more like `hitplusperc += skill_lv;` for TF_DOUBLE and for BS_WEAPONRESEARCH it's `hitplusperc += skill_lv * 2;` (with hitplusperc being initially `0`)

And that is applied at the very end of the hitrate calculation. like so:
`hitrate += hitrate * hitplusperc / 100`

(although in aegis it's more like `hitrate = hitrate * (100 + hitplusperc) / 100;`)